ctest.h

```diff
@@ -38,16 +38,23 @@ typedef void (*ctest_teardown_func)(void*);
     CTEST_IMPL_PRAGMA(GCC diagnostic ignored "-W" #w)
 #define CTEST_IMPL_DIAG_POP() \
     CTEST_IMPL_PRAGMA(GCC diagnostic pop)
+#ifndef __unused
+#define __unused	__attribute__((__unused__))
+#endif
 #else
 /* the push/pop functionality wasn't in gcc until 4.6, fallback to "ignored"  */
 #define CTEST_IMPL_DIAG_PUSH_IGNORED(w) \
     CTEST_IMPL_PRAGMA(GCC diagnostic ignored "-W" #w)
 #define CTEST_IMPL_DIAG_POP()
+#define __unused
 #endif
 #else
 /* leave them out entirely for non-GNUC compilers  */
 #define CTEST_IMPL_DIAG_PUSH_IGNORED(w)
 #define CTEST_IMPL_DIAG_POP()
+#ifdef _MSC_VER >= 1200
```

Comment:
Using undefined macro (yes, there're compilers which don't emulate GNU C compiler and are not MSVC at the same time) will potentially result in another warning which is not what you want in a PR that fixes a warning.